### PR TITLE
Warnings are error

### PR DESCRIPTION
**Pull request by smtakeda**
- Enabled -Werror to cause error on warning
- Fixed more warnings for clang

### CODE REVIEW

File: cpp/crypto/Cryptor.cpp [hunk 1]
```diff
@@ -249,6 +249,7 @@ const EVP_CIPHER *getCipher(const CryptoAlgo algo,
                    static_cast<int>(mode));*/
 
 #undef SF_CRYPTO_GET_CIPHER
+  return nullptr;
```
[the remark this answers is not in the document]
Comment (howryu) [reply cpp/crypto/Cryptor.cpp]: It's fine.

**Comment by howryu** [comment]
LGTM

**Comment by smtakeda** [comment]
@notkriswagner please go ahead to merge the change if you are ok with the changes.

**Comment by notkriswagner** [comment]
LGTM

**Pull request merged**